native/acter/src/api/account/three_pid.rs

```diff
@@ -1,31 +1,78 @@
 use acter_core::events::three_pid::{ThreePidContent, ThreePidRecord};
 use anyhow::{bail, Context, Result};
-use matrix_sdk::{
-    reqwest::{ClientBuilder, StatusCode},
-    Account,
-};
-use ruma::{thirdparty::Medium, uint, ClientSecret, SessionId};
+use matrix_sdk::reqwest::{ClientBuilder, StatusCode};
+use ruma::{assign, uint};
 use ruma_client_api::{
-    account::ThirdPartyIdRemovalStatus,
+    account::{request_3pid_management_token_via_email, ThirdPartyIdRemovalStatus},
     uiaa::{AuthData, Password, UserIdentifier},
 };
+use ruma_common::{
+    thirdparty::Medium, ClientSecret, MilliSecondsSinceUnixEpoch, OwnedClientSecret, SessionId,
+};
 use serde::Deserialize;
 use std::{collections::BTreeMap, ops::Deref};
 
-use super::{client::Client, RUNTIME};
+use super::Account;
+use crate::{api::common::clearify_error, RUNTIME};
 
 #[derive(Clone)]
-pub struct ThreePidManager {
-    account: Account,
-    client: Client,
+pub struct ThreePidEmailTokenResponse {
+    inner: request_3pid_management_token_via_email::v3::Response,
+    client_secret: OwnedClientSecret,
+}
+
+impl ThreePidEmailTokenResponse {
+    pub fn sid(&self) -> String {
+        self.inner.sid.to_string()
+    }
+
+    pub fn submit_url(&self) -> Option<String> {
+        self.inner.submit_url.clone()
+    }
+
+    pub fn client_secret(&self) -> String {
+        self.client_secret.to_string()
+    }
 }
 
-impl ThreePidManager {
+#[derive(Clone)]
+pub struct ThreePid {
+    address: String,
+    medium: Medium,
+    added_at: MilliSecondsSinceUnixEpoch,
+    validated_at: MilliSecondsSinceUnixEpoch,
+}
+
+impl ThreePid {
```

Comment:
Don't you think this name is confusing? I think it should be something like `ThreePidEntry` or `ThreePidAccount` or just `ThirdPartyId` (which the 3pid comes from) or `ExternalId` ... no?

Reply:
Frankly, I just wanted to call it `ThreePid` because `get_3pids()` returns array of them.
Sorry to confuse you.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1896/commits/b6f0caa06e3e704a674318942030b2aedf89d02a